src/context/addToWishlist.js

```diff
@@ -0,0 +1,16 @@
+import { createContext, useContext, useReducer } from "react";
+import wishlistReducer, { initialWishlist } from "reducers/wishlistReducer";
+
+const WishlistContext = createContext();
+
+const useWishlist = () => useContext(WishlistContext)
+
+const WishlistProvider = ({ children }) => {
+    return (
+        <WishlistContext.Provider value={useReducer(wishlistReducer, initialWishlist)}>
+            {children}
+        </WishlistContext.Provider>
```

Comment:
you are calling useReducer in values, it not good practice. Call it saperately on top.